sdk/tests/deqp/functional/gles3/es3fFboColorbufferTests.js

```diff
@@ -0,0 +1,979 @@
+        this.checkError();
+        this.checkFramebufferStatus(gl.FRAMEBUFFER);
+
+        ctx.viewport(0, 0, this.m_size[0], this.m_size[1]);
+
+        // Fill framebuffer with grid pattern.
+        var format = gl.RGBA;
+        var dataType = gl.UNSIGNED_BYTE;
+        var texW = 128;
+        var texH = 128;
+        var data = new tcuTexture.TextureLevel(gluTextureUtil.mapGLTransferFormat(format, dataType), texW, texH, 1);
+
+        tcuTextureUtil.fillWithGrid(data.getAccess(), 8, [0.2, 0.7, 0.1, 1.0], [0.7, 0.1, 0.5, 0.8]);
+
+        var gridTex = ctx.createTexture();
+        ctx.bindTexture(gl.TEXTURE_2D, gridTex);
+        ctx.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+        ctx.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+        ctx.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.LINEAR);
+        ctx.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.LINEAR);
+        ctx.texImage2D(gl.TEXTURE_2D, 0, format, texW, texH, 0, format, dataType, data.getAccess().getDataPtr());
+
+        rrUtil.drawQuad(ctx, texShaderID, [-1.0, -1.0, 0.0], [1.0, 1.0, 0.0]);
+        this.checkError();
+
+        // Setup blend.
+        ctx.enable(gl.BLEND);
+        ctx.blendEquationSeparate(this.m_funcRGB, this.m_funcAlpha);
+        ctx.blendFuncSeparate(this.m_srcRGB, this.m_dstRGB, this.m_srcAlpha, this.m_dstAlpha);
+
+        // Render gradient with blend.
+        rrUtil.drawQuad(ctx, gradShaderID, [-1.0, -1.0, 0.0], [1.0, 1.0, 0.0]);
+
+        es3fFboTestUtil.readPixels(ctx, dst, 0, 0, this.m_size[0], this.m_size[1], fboFmt, [1, 1, 1, 1], [0, 0, 0, 0]);
+    };
+
+/**
+ * @constructor
+ * @extends {tcuTestCase.DeqpTest}
+ */
+es3fFboColorbufferTests.FboColorbufferTests = function() {
+    tcuTestCase.DeqpTest.call(this, 'color', 'Colorbuffer tests');
+};
+
+setParentClass(es3fFboColorbufferTests.FboColorbufferTests, tcuTestCase.DeqpTest);
+
+es3fFboColorbufferTests.FboColorbufferTests.prototype.init = function() {
+    var colorFormats = [
+        // RGBA formats
+        gl.RGBA32I,
+        gl.RGBA32UI,
+        gl.RGBA16I,
+        gl.RGBA16UI,
+        gl.RGBA8,
+        gl.RGBA8I,
+        gl.RGBA8UI,
+        gl.SRGB8_ALPHA8,
+        gl.RGB10_A2,
+        gl.RGB10_A2UI,
+        gl.RGBA4,
+        gl.RGB5_A1,
+
+        // RGB formats
+        gl.RGB8,
+        gl.RGB565,
+
+        // RG formats
+        gl.RG32I,
+        gl.RG32UI,
+        gl.RG16I,
+        gl.RG16UI,
+        gl.RG8,
+        gl.RG8I,
+        gl.RG8UI,
+
+        // R formats
+        gl.R32I,
+        gl.R32UI,
+        gl.R16I,
+        gl.R16UI,
+        gl.R8,
+        gl.R8I,
+        gl.R8UI,
+
+        // gl.EXT_color_buffer_float
+        gl.RGBA32F,
+        gl.RGBA16F,
+        gl.R11F_G11F_B10F,
+        gl.RG32F,
+        gl.RG16F,
+        gl.R32F,
+        gl.R16F,
+
+        // gl.EXT_color_buffer_half_float
+        gl.RGB16F
```

Comment:
Hmm, that's unfortunate. The plan has been to not expose EXT_color_buffer_half_float in WebGL 2.0 contexts, only EXT_color_buffer_float. I seem to recall that it came up in previous working group discussions that RGB textures (either 32F or 16F) aren't made renderable in EXT_color_buffer_float. Anyway, not a big deal, and thanks for porting all of the code paths.